packages/medusa/src/services/fulfillment.ts

```diff
@@ -283,13 +292,25 @@ class FulfillmentService extends TransactionBaseService {
 
       const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
 
-      for (const fItem of fulfillment.items) {
-        const item = await lineItemServiceTx.retrieve(fItem.item_id)
-        const fulfilledQuantity = item.fulfilled_quantity! - fItem.quantity
-        await lineItemServiceTx.update(item.id, {
-          fulfilled_quantity: fulfilledQuantity,
+      const productVariantInventoryServiceTx =
+        this.productVariantInventoryService_.withTransaction(manager)
+
+      await Promise.all(
+        fulfillment.items.map(async (fItem) => {
+          const item = await lineItemServiceTx.retrieve(fItem.item_id)
+          const fulfilledQuantity = item.fulfilled_quantity! - fItem.quantity
+          await lineItemServiceTx.update(item.id, {
+            fulfilled_quantity: fulfilledQuantity,
+          })
+          if (item.variant_id) {
+            await productVariantInventoryServiceTx.adjustInventory(
+              item.variant_id,
+              fulfillment.location_id!,
+              fItem.quantity
+            )
+          }
```

Comment:
thought: this should go in the API level, right? This way we can construct the transaction orchestrator so that if one of the adjustInventory attempts fails we roll back everything